tests/test_dipole.py

```diff
@@ -155,3 +155,17 @@ def test_dipole_bmadx_tracking(dtype):
         rtol=1e-14 if dtype == torch.float64 else 0.00001,
         atol=1e-14 if dtype == torch.float64 else 1e-6,
     )
+
+
+def test_buffer_registration():
+    """
+    Test that buffers are properly registered in the dipole element.
+    """
+    length = torch.tensor(0.5)
+    gap_exit = torch.tensor(0.1)
+    dipole = Dipole(length=length, gap_exit=gap_exit)
+
+    # length and gap_exit are tested because they behave slightly different than other
+    # properties. length is inherited and gap_exit is derived from gap by default.
+    assert length in dipole.buffers()
+    assert gap_exit in dipole.buffers()
```

Comment:
I'm a little confused by this test. It only tests the "weird" buffers? Then this should be reflected in the docstring and name (I guess by saying in the docstring what it says in the comment).

Reply:
In principle I would have liked to test all buffers, but as we said last week there does not seem to be a good way to achieve this automatically. Therefore I selected the two arguments that seemed most likely to cause trouble to me. We could of course extend this to cover more buffers of `Dipole`.

Reply:
@Hespe, what do you think about the `test_buffer_registration` test now? I just added all buffers. I think because the test is purely for `Dipole` it's fair to handcode what we would expect.

Reply:
Seems reasonable. But its something to keep in mind should we need to add more arguments/buffers later.

Reply:
Okay, I've added a check for the number of buffers. If we add stuff, there is a good chance this will change and alert us to update the test.